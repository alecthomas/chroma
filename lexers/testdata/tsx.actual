import React from 'react'
import ReactDOM from 'react-dom'
import './index.css'
import { App } from './App'

ReactDOM.render(
  <>
    <React.StrictMode>
      <App />
    </React.StrictMode>
  </>,
  document.getElementById('root'),
)

ReactDOM.render(
  <p
  // This is a inline comment.
  /*
    This is a
    Multi line comment
  */
  <!-- Another multiline
  comment !-->
  data-test-id="outro" disabled>
	Some text here.
  </p>,
  document.getElementById('root'),
)