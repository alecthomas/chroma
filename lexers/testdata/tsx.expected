[
  {"type":"KeywordReserved","value":"import"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"React"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"from"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringSingle","value":"'react'"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordReserved","value":"import"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"ReactDOM"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"from"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringSingle","value":"'react-dom'"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordReserved","value":"import"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringSingle","value":"'./index.css'"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordReserved","value":"import"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"App"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"}"},
  {"type":"Text","value":" "},
  {"type":"NameOther","value":"from"},
  {"type":"Text","value":" "},
  {"type":"LiteralStringSingle","value":"'./App'"},
  {"type":"Text","value":"\n\n"},
  {"type":"NameOther","value":"ReactDOM"},
  {"type":"Punctuation","value":"."},
  {"type":"NameOther","value":"render"},
  {"type":"Punctuation","value":"("},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"\u003c\u003e"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":"\u003c"},
  {"type":"NameTag","value":"React.StrictMode"},
  {"type":"Punctuation","value":"\u003e"},
  {"type":"Text","value":"\n      "},
  {"type":"Punctuation","value":"\u003c"},
  {"type":"NameTag","value":"App"},
  {"type":"Text","value":" "},
  {"type":"Punctuation","value":"/\u003e"},
  {"type":"Text","value":"\n    "},
  {"type":"Punctuation","value":"\u003c/"},
  {"type":"NameTag","value":"React.StrictMode"},
  {"type":"Punctuation","value":"\u003e"},
  {"type":"Text","value":"\n  "},
  {"type":"Punctuation","value":"\u003c/\u003e,"},
  {"type":"Text","value":"\n  "},
  {"type":"NameBuiltin","value":"document"},
  {"type":"Punctuation","value":"."},
  {"type":"NameOther","value":"getElementById"},
  {"type":"Punctuation","value":"("},
  {"type":"LiteralStringSingle","value":"'root'"},
  {"type":"Punctuation","value":"),"},
  {"type":"Text","value":"\n"},
  {"type":"Punctuation","value":")"},
  {"type":"Text","value":"\n"}
]
